### PR TITLE
refactor: use instanceof

### PR DESCRIPTION
**Pull request by ursjoss**
Netbeans refactoring to avoid the casting. Plus occasional rename of the automatically created variable.

If we want to go this way, we should merge #102 first and rebase this PR onto main afterwards.

- [x] Decide if this is the path to go
- [x] Test and merge #102
- [x] Rebase this branch onto main

**Note:** This PR requires a language feature from java 17!

### CODE REVIEW

**Comment by ursjoss** [comment]
Trying to use a java17 languate features while compiling for java 11. This doesn't work.

File: modules/au.com.trgtd.tr.view.goals/src/au/com/trgtd/tr/view/goals/levels/LevelNodeRoot.java [hunk 1]
```diff
@@ -85,8 +85,7 @@ public boolean canCut() {
     @Override
     public PasteType getDropType(Transferable t, final int action, int index) {
         final Node dropNode = NodeTransfer.node(t, DnDConstants.ACTION_COPY_OR_MOVE + NodeTransfer.CLIPBOARD_CUT);
-        if (dropNode instanceof LevelNode) {
-            final LevelNode dropLevelNode = (LevelNode) dropNode;
```
[the remark this answers is not in the document]
Comment (RadekCap) [reply modules/au.com.trgtd.tr.view.goals/src/au/com/trgtd/tr/view/goals/levels/LevelNodeRoot.java:89]: :) I see, my bad, we are preserving final here, of course.

**Review by RadekCap — APPROVED** [review]
lgtm, ty.

We have to re-solve issues with recently reintroduced module load mess. #64

**Comment by ursjoss** [comment]
Thanks a lot @RadekCap

**Pull request merged**